go.mod

```diff
@@ -132,7 +132,7 @@ require (
 	k8s.io/klog/v2 v2.130.1
 	k8s.io/kubectl v0.32.1
 	k8s.io/utils v0.0.0-20241210054802-24370beab758
-	sigs.k8s.io/controller-runtime v0.19.4
+	sigs.k8s.io/controller-runtime v0.20.1
```

Comment:
sigs.k8s.io/* auto update is disabled as part of https://github.com/cilium/cilium/blob/da69c875bef922c4d0dae571e8639863a94cade6/.github/renovate.json5#L558-LL569, we probably should update the k8s upgrade docs later to bump this library.